### ISSUE TITLE
NTFS kernel-mode driver support

### ISSUE THREAD

**Issue by HanabishiRecca** [storaged-project/udisks#798]
Paragon Software submitted a NTFS kernel-mode driver to the mainline Linux kernel.  
https://www.phoronix.com/scan.php?page=news_item&px=Paragon-Read-Write-NTFS-Linux  
https://www.phoronix.com/scan.php?page=news_item&px=Paragon-NTFS3-V3-For-Linux  

But `udisks` can't use kernel-mode NTFS drivers without an explicit fs type declaration. For example:   

`udisksctl mount -b /dev/sde1` - raises an error.  
```
Error mounting /dev/sde1: GDBus.Error:org.freedesktop.UDisks2.Error.Failed: Error mounting /dev/sde1 at /run/media/hanabishi/Toshiba P300: wrong fs type, bad option, bad superblock on /dev/sde1, missing codepage or helper program, or other error
```
Obviously it wants a FUSE driver here.  

`udisksctl mount -b /dev/sde1 -t ntfs3` - works with the explicit fs type.  

The problem is 3rd party software like GIO, file managers, disk managers etc. obviously does not specifying the fs type.  
So now when the NTFS kernel-mode driver exists (and possibly can be accepted to the mainline Linux kernel), I think the default support of it should be implemented.

**Comment by tbzatek** [comment]
Well I think it's up to the kernel to decide which driver to use. There's the legacy ntfs one which I expect the new `ntfs3` will seamlessly replace. Let's wait until it gets fully merged, what fs type would it use. Need to coordinate with `util-linux`/`libmount` as well.

**Comment by HanabishiRecca** [comment]
@tbzatek maybe you are right. For now I'm using such workaround (udev rule):  
`SUBSYSTEM=="block", ENV{ID_FS_TYPE}=="ntfs", ENV{ID_FS_TYPE}="ntfs3"`

**Comment by vojtechtrefny** [comment]
Does the `mount` command work (without specifying type)? We are using libmount so UDisks should behave the same (although we are passing type to libmount, `ntfs` in this case, but it's up to libmount to decide whether to use a kernel driver or a "helper" program like `mount.ntfs-3g`).

**Comment by HanabishiRecca** [comment]
@vojtechtrefny no, it also can't pick up the new driver automatically. Mount tries to use the legacy ntfs kernel driver or fails with a unknown filesystem error when default ntfs kmod is disabled.
I just thought udisks handles mounting in a more complex way. But if it's just up to libmount behavior, alright then, I hurried to post this issue here.

The good thing is, seems like udisks works fine with ntfs3 itself using the udev trick. So maybe you won't need fix anything in the future at all.

**Comment by vojtechtrefny** [comment]
We usually do not try to be smarter than libmount (because we are not) but we can definitely revisit this after the driver is merged into mainline kernel. Libmount has na API to force use the kernel driver so we can do something like check kernel version or driver availability and force it.

Glad to hear the udev workaround works for you. Maybe we could add similar configuration option to the mount options config to make it even easier in the future, thought @tbzatek?

**Comment by HanabishiRecca** [comment]
One more thing: udisks obviously has no inbuilt rules for `ntfs3`. So it mounts fs as root and without default ntfs flags.
I've added ntfs3 rules in the `mount_options.conf` (just copied from default ntfs).
```
ntfs3_defaults=uid=$UID,gid=$GID,windows_names
ntfs3_allow=uid=$UID,gid=$GID,umask,dmask,fmask,locale,norecover,ignore_case,windows_names,compression,nocompression,big_writes,user,users
```

**Comment by JpegXguy** [comment]
IT seems like it will remain named "ntfs3" for now
https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/commit/?id=f7464060f7ab9a2424428008f0ee9f1e267e410f

**Comment by tbzatek** [comment]
So I've disabled the legacy kernel `ntfs` driver, uninstalled `ntfs-3g` and any attempt to mount `ntfs` filesystem ends up with `unknown filesystem type 'ntfs'.` error. The `ntfs3` filesystem type must be specified manually, at least at this point.

**Comment by JpegXguy** [comment]
I'm using udisks 2.9.4-1 on arch and I have both ntfs-3g and ntfs3 available because I'm using Linux 5.15.
I though udisks would prefer ntfs3 over ntfs but my ntfs disk is still mounted though fuse

**Comment by HanabishiRecca** [comment]
@JpegXguy, already discussed it in PR #917
